server/http_test.go

```diff
@@ -1079,135 +1070,99 @@ func TestEditSubscriptionTemplate(t *testing.T) {
-								IssueTypes: NewStringSet("10001"),
-							},
+							ID:      "mockTemplateID1___________",
+							Filters: getMockSubscriptionFilter("jira:issue_created"),
 						},
 					}), t),
 		},
 		"Editing subscription template, no issue types provided": {
-			subscriptionTemplate:       `{"instance_id": "jiraurl1", "name": "somename", "id": "aaaaaaaaaaaaaaaaaaaaaaaaab", "channel_id": "aaaaaaaaaaaaaaaaaaaaaaaaac", "filters": {"events": ["jira:issue_created"], "projects": ["otherproject"], "issue_types": []}}`,
-			expectedStatusCode: http.StatusInternalServerError,
+			subscriptionTemplate: `{"instance_id": "jiraurl1", "name": "somename", "id": "mockTemplateID1___________", "channel_id": "aaaaaaaaaaaaaaaaaaaaaaaaac", "filters": {"events": ["jira:issue_created"], "projects": ["otherproject"], "issue_types": []}}`,
+			expectedStatusCode:   http.StatusInternalServerError,
 			apiCalls: checkHasSubscriptionTemplates([]SubscriptionTemplate{},
 				withExistingChannelSubscriptionTemplates(
 					[]SubscriptionTemplate{
 						{
-							ID: "aaaaaaaaaaaaaaaaaaaaaaaaab",
-							Filters: SubscriptionFilters{
-								Events:     NewStringSet("jira:issue_created"),
-								Projects:   NewStringSet("myproject"),
-								IssueTypes: NewStringSet("10001"),
-							},
+							ID:      "mockTemplateID1___________",
+							Filters: getMockSubscriptionFilter("jira:issue_created"),
 						},
 					}), t),
 		},
-		"Editing subscription template, GetProject mocked error. Existing sub has nonexistent project.": {
-			subscriptionTemplate:       fmt.Sprintf(`{"instance_id": "jiraurl1", "id": "subaaaaaaaaaabbbbbbbbbbccc", "name": "subscription name", "channel_id": "channelaaaaaaaaaabbbbbbbbb", "filters": {"events": ["jira:issue_created"], "projects": ["%s"], "issue_types": ["10001"]}}`, nonExistantProjectKey),
-			expectedStatusCode: http.StatusInternalServerError,
+		"Editing subscription template, GetProject mocked error. Existing subscription has a non-existent project.": {
+			subscriptionTemplate: fmt.Sprintf(`{"instance_id": "jiraurl1", "id": "mockTemplateID2___________", "name": "subscription name", "channel_id": "channelaaaaaaaaaabbbbbbbbb", "filters": {"events": ["jira:issue_created"], "projects": ["%s"], "issue_types": ["10001"]}}`, nonExistantProjectKey),
+			expectedStatusCode:   http.StatusInternalServerError,
 			apiCalls: checkHasSubscriptionTemplates([]SubscriptionTemplate{},
 				withExistingChannelSubscriptionTemplates(
 					[]SubscriptionTemplate{
 						{
-							ID: "subaaaaaaaaaabbbbbbbbbbccc",
-							Filters: SubscriptionFilters{
-								Events:     NewStringSet("jira:issue_updated"),
-								Projects:   NewStringSet("myproject"),
-								IssueTypes: NewStringSet("10001"),
-							},
+							ID:      "mockTemplateID2___________",
+							Filters: getMockSubscriptionFilter("jira:issue_updated"),
 						},
 					}), t),
 		},
 		"Editing subscription template, GetProject mocked error. Existing sub has existing project.": {
```

Comment:
Editing subscription template, GetProject mocked error. Existing subscription has the existing project.